lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -634,22 +698,26 @@ void LowerLayersPass::runOnOperation() {
   // Initialize members which cannot be initialized automatically.
   llvm::sys::SmartMutex<true> mutex;
   circuitMutex = &mutex;
-  symbolTable = &getAnalysis<SymbolTable>();
 
-  // Determine names for all modules that will be created.  Do this serially to
-  // avoid non-determinism from creating these in the parallel region.
   CircuitNamespace ns(circuitOp);
-  WalkResult result = circuitOp->walk([&](FModuleOp moduleOp) {
-    moduleOp->walk([&](LayerBlockOp layerBlockOp) {
-      auto name = moduleNameForLayer(moduleOp.getModuleName(),
-                                     layerBlockOp.getLayerName());
-      moduleNames.insert({layerBlockOp, ns.newName(name)});
-    });
-    return WalkResult::advance();
-  });
-
-  if (result.wasInterrupted())
-    return signalPassFailure();
+  for (auto &op : *circuitOp.getBodyBlock()) {
+    // Determine names for all modules that will be created.  Do this serially
+    // to avoid non-determinism from creating these in the parallel region.
+    if (auto moduleOp = dyn_cast<FModuleOp>(op)) {
+      moduleOp->walk([&](LayerBlockOp layerBlockOp) {
+        auto name = moduleNameForLayer(moduleOp.getModuleName(),
+                                       layerBlockOp.getLayerName());
+        moduleNames.insert({layerBlockOp, ns.newName(name)});
+      });
+      continue;
+    }
+    // Build verilog macro declarations for each inline layer declarations.
+    if (auto layerOp = dyn_cast<LayerOp>(op)) {
+      OpBuilder b(layerOp);
+      createMacroDecls(ns, b, layerOp);
+      continue;
+    }
+  }
```

Comment:
This would be more future-proof as a walk or something which does not assume that `LayerOp` is at the top-level of a module.